pkg/sidecar/pod.go

```diff
@@ -39,7 +40,7 @@ func add(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCo
 		return pod, err
 	}
 
-	container := collector.Container(cfg, logger, otelcol, false)
+	container := collector.Container(cfg, logger, convert.V1Alpha1to2(otelcol), false)
```

Comment:
Shouldn't the signature of `add` change to the new type?

Reply:
I plan to change that in one of the follow up prs.
In this one I tried to limited the scope only to `internal/manifests/collector`.